app/routes/_public.allagan-data.tsx

```
@@ -99,12 +110,32 @@
+  const discordId = session.get('discord_id')
+  const discordRoles = session.get('discord_roles') || []
+  const isLoggedIn = !!discordId
+  const hasPremium = getHasPremium(discordRoles)
+
+  return json({
+    isLoggedIn,
+    hasPremium
+  })
+}
+
 type ActionResponse = AllaganResults | { exception: string } | {}
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)